.github/workflows/generate-preview.yml

```diff
@@ -0,0 +1,93 @@
+name: Generate asciinema preview gif
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  generate:
+    runs-on: ubuntu-latest
+    env:
+      DEMO_MAGIC_INSTALL: /tmp/demo-magic
+      TERM: xterm
+    steps:
+    # Checks-out the project under $GITHUB_WORKSPACE/git-gud
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        path: git-gud
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '14'
+
+    - name: Set up packager-managed dependencies for gif build
+      # demo-magic.sh requires [pv]
+      run: |
+        sudo apt-get update
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine
+        pip install $GITHUB_WORKSPACE/git-gud
+        pip install asciinema
+        sudo apt-get install pv
+
+    - name: Download demo-magic.sh from repository
+      run: git clone --depth 1 https://github.com/paxtonhare/demo-magic.git $DEMO_MAGIC_INSTALL
+
+    - name: Test functionality of preview commands
+      run: |
+        chmod +x $GITHUB_WORKSPACE/git-gud/previewgif.sh
+        $GITHUB_WORKSPACE/git-gud/previewgif.sh $GITHUB_WORKSPACE/git-gud
+    
+    - name: Record to asciinema cast file (JSON)
+      run: |
+        python3 -m asciinema rec -c "$GITHUB_WORKSPACE/git-gud/previewgif.sh $GITHUB_WORKSPACE/git-gud" -e SHELL,TERM,DEMO_MAGIC_INSTALL asciicast.json
+
+    - name: Use dgzlopes/asciicast-to-gif-action@v1.0
+      uses: dgzlopes/asciicast-to-gif-action@v1.0
+      with:
+        input: asciicast.json
+        output: preview.gif
+
+    - name: Move everything to build
+      run: |
+        mkdir /tmp/build
+        cp ./asciicast.json /tmp/build
+        cp ./preview.gif /tmp/build
+
+    - name: Upload asciicast.json
+      uses: actions/upload-artifact@v2
+      with:
+        name: asciicast.json
+        path: /tmp/build/asciicast.json
```

Comment:
Why do we upload this one?

Reply:
This is uploaded as an artifact and is not ever committed to the repository. *Ever.*

This is useful if we ever decide that we want the asciicast.json file for inspection, testing alternatives to asciicast2gif, and more.